tests/Spork/Test/Batch/BatchJobTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Spork\Test\Batch;
+
+class BatchJobTest extends \PHPUnit_Framework_TestCase
+{
+    public function testErrorPropagation()
+    {
+        $manager = new \Spork\ProcessManager();
+        $batch = $manager->createBatchJob(range(1, 5));
+
+        $expectedExitStatus = 20;
+
+        $failingClosure = function ($data) use ($expectedExitStatus) {
+            // Simple condition to simulate only one point of failure
+            if (3 === $data) {
+                exit($expectedExitStatus);
+            }
+
+            exit(0);
+        };
+
+        $promise = $batch->execute($failingClosure);
+
+        $promise->wait();
+
+        $success = null;
+
+        $promise->done(function () use (& $success) {
+            $success = true;
+        });
+
+        $promise->fail(function () use (& $success) {
+            $success = false;
+        });
+
+        $this->assertFalse($success, 'Promise should fail');
+
+        $this->assertNull($promise->getError(), 'Child process did not throw and exception so error should be null');
+
+        $actualExitStatus = $promise->getExitStatus();
+
+        $this->assertEquals($expectedExitStatus, $actualExitStatus, 'Parent process exit status should match the child one');
+    }
+
+    public function testErrorMessagePropagation()
+    {
+        $manager = new \Spork\ProcessManager();
+        $batch = $manager->createBatchJob(range(1, 5));
+
+        $expectedErrorMessage = 'This is a test message';
+
+        $failingClosure = function ($data) use ($expectedErrorMessage) {
+            // Simple condition to simulate only one point of failure
+            if (3 === $data) {
+                throw new \InvalidArgumentException($expectedErrorMessage);
+            }
+
+            exit(0);
```

Comment:
do we absolutely need to exit here ? It would be great if this example could match what people will probably do, i.e. not using ``exit`` at all